engine/src/main/java/org/pentaho/di/base/AbstractMeta.java

```diff
@@ -139,13 +139,13 @@
 
   protected List<DatabaseMeta> databases;
 
-  protected List<NameChangedListener> nameChangedListeners;
+  protected Set<NameChangedListener> nameChangedListeners = Collections.newSetFromMap( new ConcurrentHashMap<NameChangedListener, Boolean>() );
```

Comment:
Sets are backed by the map, not copied, so this is OK.

https://docs.oracle.com/javase/7/docs/api/java/util/Collections.html#newSetFromMap(java.util.Map)